Algorithm/QCAlgorithm.History.cs

```diff
@@ -1141,9 +1141,15 @@ private IEnumerable<SubscriptionDataConfig> GetMatchingSubscriptions(Symbol symb
                     var isCustom = Extensions.IsCustomDataType(symbol, type);
                     var entry = MarketHoursDatabase.GetEntry(symbol, new[] { type });
 
+                    // If the type is PythonData, we want to check what types are actually associated with the symbol
+                    if (!_customPythonDataTypes.TryGetValue(symbol, out var dataType))
```

Comment:
Shouldn't the dataType already be available in an existing config above, believe handling those correctly might avoid the need this new cache